app/models/product_drive.rb

```diff
@@ -46,8 +46,14 @@ def donation_quantity
     donations.joins(:line_items).sum('line_items.quantity')
   end
 
-  def distinct_items
-    donations.joins(:items).distinct(:item_id).count
+  def distinct_items_count
+    item_ids = []
+    donations.each do |donation|
+      donation.line_items.each do |line_item|
+        item_ids << line_item.item_id
+      end
+    end
```

Comment:
This can be done in a bit more performant manner. Pretty sure this should work:
```ruby
self.donations.
  joins(:line_items).
  select('line_items.item_id').
  distinct.
  count
```

Reply:
It did !